User Manual

KtorZ opened the pull request:
![Screen Shot 2022-03-09 at 18 14 50](https://user-images.githubusercontent.com/5680256/157497662-aeb8693e-a801-467b-b781-ddd3987161a7.png)

---

![Screen Shot 2022-03-09 at 18 14 59](https://user-images.githubusercontent.com/5680256/157497656-7d5aa2b2-f4d3-49e5-8a3d-8e4617173931.png)

---

![Screen Shot 2022-03-09 at 18 14 54](https://user-images.githubusercontent.com/5680256/157497659-3ee7fe55-1cae-4c75-98f6-85f1dfffdf55.png)

---

![Screen Shot 2022-03-09 at 18 15 06](https://user-images.githubusercontent.com/5680256/157497652-a0136eb5-ea88-41d6-961b-64464730f604.png)

---

![Screen Shot 2022-03-09 at 18 15 16](https://user-images.githubusercontent.com/5680256/157497650-84f74f77-10ff-419f-9de3-895a39a6976d.png)

---

![Screen Shot 2022-03-09 at 18 15 43](https://user-images.githubusercontent.com/5680256/157497639-bc9af511-0caf-4e63-b659-f4fc06d93201.png)

---

![Screen Shot 2022-03-09 at 18 15 39](https://user-images.githubusercontent.com/5680256/157497645-e644bbf6-e435-4094-9d24-8180070fd7d5.png)

---

<img width=220 src="https://user-images.githubusercontent.com/5680256/157497665-c875a3aa-a7d4-4f1a-be0c-d2e7be27fc45.png" />  <img width=220 src="https://user-images.githubusercontent.com/5680256/157497668-974ee7db-2a8f-474f-893e-d307f8f12969.png"/>  <img width=220 src="https://user-images.githubusercontent.com/5680256/157497670-bbaa2dee-c289-4fe8-81be-fbfeb99f0fe5.png"/>

KtorZ committed:
Setup docusaurus documentation.

KtorZ committed:
Convert and bundle existing documentation into the user-manual.

KtorZ committed:
Replace existing doc with user-manual.

KtorZ committed:
Adjust CI workflow to build documentation with Docusaurus.

KtorZ committed:
Delete authors.yml

ch1bo commented on demo/README.md:
Should: update the link in the main `README` to point directly into the user guide. The indirection here is probably not needed.

ch1bo commented on docs/README.md:
So is `docs/` a website now? Can you maybe add another sentence here where our docs website is located now and this is the instructions to build/hack on them would make this a bit less generic :)

ch1bo commented on docs/adr/2021-11-25_012-top-down-test-driven-design.md:
```diff
@@ -1,10 +1,14 @@
+title: | 
+  12. Top-down Test-driven Design
+authors: []
+tags: [Draft]
```
Nice :+1: Another reason to get them more accepted now :)

KtorZ commented on demo/README.md:
Yes indeed. This emoji does not appear on Mac it seems.

See why I hesitate so much for a Mac... They lack the good stuff!

KtorZ committed:
Tweak demo & docs READMEs.

KtorZ committed:
Bootstrap API reference page.

ch1bo reviewed (approved):
Just realized I had these comments pending. Still approve, lets merge and polish it as we go!

KtorZ committed:
Fix dark theme for the api doc and fix rendering of script from within React.

  I think I just hate frontend JavaScript. It's good to get a reminder from time to time.

KtorZ committed:
Translate existing JSON-schema to async-api format.

  Now though, we need to adjust the tests to use that new format and, get rid of the duplication.

KtorZ commented:
Added support for https://www.asyncapi.com/ and translated our existing JSON-schema to it. Pretty neat.

![Screenshot from 2022-03-10 18-59-04](https://user-images.githubusercontent.com/5680256/157726291-d069c28a-668e-45a1-bd39-3bac10ab07cc.png)

![Screenshot from 2022-03-10 18-59-16](https://user-images.githubusercontent.com/5680256/157726299-ae57e6ca-048c-4f69-9a54-1d003ca9742a.png)

KtorZ committed:
Remove 'docs' from checked markdown links workflow.

  Already checked by Docusaurus.

KtorZ committed:
Fix dead links in CHANGELOG.

KtorZ committed:
Adjust color-scheme and illustrations. Now we're ready.

ch1bo committed:
Add yarn to shell.nix

ch1bo commented:
> Added support for https://www.asyncapi.com/ and translated our existing JSON-schema to it. Pretty neat.

Is this already scope of this? In the generated website the API link was still a TODO.

KtorZ commented:
@ch1bo I did replace the link from the config yesterday and I've been using the top bar to access it many times already.

Maybe I forgot to commit that 🤔?

KtorZ committed:
Fixup setup Node.js workflow.

KtorZ committed:
Shorten root README, removing now-redundant sections and pointing to the user manual instead.

KtorZ committed:
Move haskell packages from README into a new doc section.

KtorZ committed:
Revert "Add yarn to shell.nix"

  This reverts commit 4c42f452c14777312b64781080da6c6e1a87f516.
  I believe that this is causing a cache invalidation on CI, which is
  causing a rebuild of the entire universe... While this is useful and
  will be necessary eventually, it is right now annoying as I am trying
  to get the CI workflow operational. So, will re-add afterwards once
  the whole pipeline is working.
